### ISSUE TITLE
Issue running the books-ms jenkins job 

### ISSUE THREAD

**Issue by naslanidis** [vfarcic/books-ms#45]
In the Continuous Integration, Delivery and Deployment chapter I am having issues running the jenkins jobs:

[Pipeline] stage (Build service)
Using the ‘stage’ step without a block argument is deprecated
Entering stage Build service
Proceeding
[Pipeline] sh
[books-ms] Running shell script
+ docker pull 10.100.198.200:5000/books-ms
Using default tag: latest
latest: Pulling from books-ms
Digest: sha256:456358ca46d2213cb3f5cef50543f52282190e07192e2119a509ead6328e5ff3
Status: Image is up to date for 10.100.198.200:5000/books-ms:latest
[Pipeline] sh
[books-ms] Running shell script
+ docker build -t 10.100.198.200:5000/books-ms .
Sending build context to Docker daemon  43.27MB

Step 1/11 : FROM java
 ---> d23bdf5b1b1b
Step 2/11 : MAINTAINER Viktor Farcic "viktor@farcic.com"
 ---> Using cache
 ---> b1912414b29c
Step 3/11 : ENV DB_DBNAME books
 ---> Using cache
 ---> e08fc9ae7ad0
Step 4/11 : ENV DB_COLLECTION books
 ---> Using cache
 ---> 43ec5ce2a87c
Step 5/11 : ENV DB_HOST localhost
 ---> Using cache
 ---> 7dcc2d32b99c
Step 6/11 : COPY run.sh /run.sh
 ---> Using cache
 ---> 5eb3249f98eb
Step 7/11 : RUN chmod +x /run.sh
 ---> Running in 49a3f07a5900
[91mcontainer_linux.go:247: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"
[0moci runtime error: container_linux.go:247: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory"

The issue is with the docker build phase and the problem exists for both the books-ms and books-ms-ansible jobs. I manually ran the docker build command that is failing and I get the same result. 

I've deleted my local images and rerun everything in the chapter with the same result as well. Any assistance or suggestions would be appreciated.

**Comment by vfarcic** [comment]
The error indicates that `books-ms-assembly-1.0.jar` file is not present. That means that the previous step that builds the JAR failed. Can you paste the whole log?

**Comment by naslanidis** [comment]
Thanks. I will get the whole log when at home, however I worked around this problem after a lot of troubleshooting by swapping the FROM java base image in the Dockerfile to FROM openjdk. I noticed that the official java docker container is depreciated and thought I'd try their recommendation of using openkdk

https://hub.docker.com/_/java/
"This image is officially deprecated in favor of the openjdk image, and will receive no further updates after 2016-12-31 (Dec 31, 2016). Please adjust your usage accordingly."

I did not see this problem again after changing to openjdk as the base image in the docker file. 

The jenkins job still fails but it's now due to the following so there is not much you can do about that:
https://issues.jenkins-ci.org/browse/JENKINS-42152

**Comment by naslanidis** [comment]
Update: 

Unfortunately I've lost the original logs as it has been working since I changed to FROM openjdk for the base image and I've done a bunch of rebuilds to troubleshoot a few other issues. I spent a long time looking troubleshooting the error and confirming nothing was missing etc. but I couldn't find anything and it's been flawless since changing the base image to the more current (openjdk) one. I'd consider making that switch yourself as the 'java' container is now depreciated. 

On the Jenkins docker workflow bug linked above I compiled my own version of the Jenkins plugin and installed it using your Ansible Jenkins role and now the books-ms job works to completion. If anyone else has a similar issue I can provide my compiled version of the plugin but there is a lot of people complaining about it and it will likely be fixed very soon as it's a simple issue (an invalid --force argument being used).
